src/cdk/dialog/dialog.ts

```diff
@@ -315,9 +316,16 @@ export class Dialog implements OnDestroy {
       (!userInjector ||
         !userInjector.get<Directionality | null>(Directionality, null, {optional: true}))
     ) {
+      const valueSignal = signal(config.direction);
       providers.push({
         provide: Directionality,
-        useValue: {value: config.direction, change: observableOf()},
+        useValue: {
```

Comment:
Maybe move this literal into a variable and type it as `Directionality` so TS surfaces any interface changes in the future?

Reply:
Done. At some point we probably should change `Directionality` to a class that can be constructed normally instead of only through DI, or maybe just change it to an interface, but this works for now.